app/src/main/java/dev/pinkroom/marketsight/ui/theme/Type.kt

```diff
@@ -1,34 +1,43 @@
+import androidx.compose.ui.text.font.FontStyle
 import androidx.compose.ui.text.font.FontWeight
-import androidx.compose.ui.unit.sp
+import dev.pinkroom.marketsight.R
 
-// Set of Material typography styles to start with
+
+val FontFamilyUbuntu = FontFamily(
+    Font(resId = R.font.ubuntu_regular, weight = FontWeight.Normal),
+    Font(resId = R.font.ubuntu_bold, weight = FontWeight.Bold),
+    Font(resId = R.font.ubuntu_bold_italic, weight = FontWeight.Bold, style = FontStyle.Italic),
+    Font(resId = R.font.ubuntu_italic, weight = FontWeight.Normal, style = FontStyle.Italic),
+    Font(resId = R.font.ubuntu_light, weight = FontWeight.Light),
+    Font(resId = R.font.ubuntu_light_italic, weight = FontWeight.Light, style = FontStyle.Italic),
+    Font(resId = R.font.ubuntu_medium, weight = FontWeight.Medium),
+    Font(resId = R.font.ubuntu_medium_italic, weight = FontWeight.Medium, style = FontStyle.Italic),
+)
+
+private val defaultTypography = Typography()
 val Typography = Typography(
-    bodyLarge = TextStyle(
-        fontFamily = FontFamily.Default,
-        fontWeight = FontWeight.Normal,
-        fontSize = 16.sp,
-        lineHeight = 24.sp,
-        letterSpacing = 0.5.sp
-    )
-    /* Other default text styles to override
-    titleLarge = TextStyle(
-        fontFamily = FontFamily.Default,
-        fontWeight = FontWeight.Normal,
-        fontSize = 22.sp,
-        lineHeight = 28.sp,
-        letterSpacing = 0.sp
-    ),
-    labelSmall = TextStyle(
-        fontFamily = FontFamily.Default,
-        fontWeight = FontWeight.Medium,
-        fontSize = 11.sp,
-        lineHeight = 16.sp,
-        letterSpacing = 0.5.sp
-    )
-    */
+    displayLarge = defaultTypography.displayLarge.copy(fontFamily = FontFamilyUbuntu),
+    displayMedium = defaultTypography.displayMedium.copy(fontFamily = FontFamilyUbuntu),
+    displaySmall = defaultTypography.displaySmall.copy(fontFamily = FontFamilyUbuntu),
+
+    headlineLarge = defaultTypography.headlineLarge.copy(fontFamily = FontFamilyUbuntu),
+    headlineMedium = defaultTypography.headlineMedium.copy(fontFamily = FontFamilyUbuntu),
+    headlineSmall = defaultTypography.headlineSmall.copy(fontFamily = FontFamilyUbuntu),
+
+    titleLarge = defaultTypography.titleLarge.copy(fontFamily = FontFamilyUbuntu),
+    titleMedium = defaultTypography.titleMedium.copy(fontFamily = FontFamilyUbuntu),
+    titleSmall = defaultTypography.titleSmall.copy(fontFamily = FontFamilyUbuntu),
+
+    bodyLarge = defaultTypography.bodyLarge.copy(fontFamily = FontFamilyUbuntu),
+    bodyMedium = defaultTypography.bodyMedium.copy(fontFamily = FontFamilyUbuntu),
+    bodySmall = defaultTypography.bodySmall.copy(fontFamily = FontFamilyUbuntu),
+
+    labelLarge = defaultTypography.labelLarge.copy(fontFamily = FontFamilyUbuntu),
+    labelMedium = defaultTypography.labelMedium.copy(fontFamily = FontFamilyUbuntu),
+    labelSmall = defaultTypography.labelSmall.copy(fontFamily = FontFamilyUbuntu)
```

Comment:
Not sure if the idea is to have specific dimensions for every typography style, but if so, we can already create our Dimens.kt file and add some default values to add in here.

Typically our file looks like this:
```kotlin
class Dimensions(
    // Padding
    val xSmallPadding: Dp = 4.dp,
    val smallPadding: Dp = 8.dp,
    val normalPadding: Dp = 16.dp,
    val largePadding: Dp = 24.dp,
    val xLargePadding: Dp = 32.dp,
    // Text
    val xSmallText: TextUnit = 12.sp,
    val smallText: TextUnit = 14.sp,
    val normalText: TextUnit = 16.sp,
    val largeText: TextUnit = 24.sp,
    val xlargeText: TextUnit = 32.sp,    
    // Etc...
)

val dimens: Dimensions
    @Composable get() {
        val configuration = LocalConfiguration.current
        return when {
            configuration.screenHeightDp <= 700 -> smallDimensions
            configuration.screenHeightDp <= 960 -> normalDimensions
            else -> largeDimensions
        }
    }

// Here you will override the dimensions needed depending on the screen size
private val smallDimensions = Dimensions()
private val normalDimensions = Dimensions()
private val largeDimensions = Dimensions()
```
You will need to adjust the file above depending on your app design guidelines.

Reply:
I know the code seems strange, but the objective is just to override the theme's font family. In material3, in the Typography object, the defaultFontFamily parameter no longer exists, so the FontFamily must be changed in all Typography members. And the only way to change it is this way. https://issuetracker.google.com/issues/236358112?pli=1